### PR TITLE
Change display of archived courses for instructor

### LINKED ISSUE #2413: Prior Term courses should be separated from current courses on Submitty main page
Right now it's a jumble.  should be:

----------

"Your courses"

Within this section the list should be grouped by role (instructor, full access grader etc)
Then sorted by semester, then by course name.  (it looks like its currently sorted by course name first?)

"Your archived courses"

same sort as above

----------

We had a design for the archive status of a course, and whether it should be hidden from students or not.  Should be stored in the master database.   @pbailie please help as needed with the master database design.

screen shot 2018-07-13 at 12 58 56 pm

### PR DESCRIPTION
**Pull request by ropern123**
Changes how archived courses are displayed by making a new "Your Archived Courses" section under the "Your Courses" section on the homepage.

Also, changed the ordering of courses when retrieving from the database.

closes #2413

### CODE REVIEW

File: site/app/libraries/database/DatabaseQueries.php [hunk 1]
```diff
@@ -1978,25 +1978,61 @@ public function insertPeerGradingAssignment($grader, $student, $gradeable_id) {
...
         $this->submitty_db->query("
 SELECT u.semester, u.course
 FROM courses_users u
 INNER JOIN courses c ON u.course=c.course AND u.semester=c.semester
-WHERE (u.user_id=? AND u.user_group=1) OR (u.user_id=? AND c.status=1)
-ORDER BY u.course", array($user_id, $user_id));
+WHERE u.user_id=? AND c.status=1
+ORDER BY u.user_group ASC,
+         SUBSTRING(u.semester, 2, 2)::INT DESC,
+         CASE WHEN SUBSTRING(u.semester, 1, 1) = 's' THEN '1'
```
Comment (bmcutler) [inline site/app/libraries/database/DatabaseQueries.php]: How does this work for other semester codes? 
(other schools may use different letters or semester labeling schemes)
Does it crash or default to alphabetic (or reverse alphabetic)?

Comment (ropern123) [reply site/app/libraries/database/DatabaseQueries.php]: No clue at the moment.
Would the desired behavior be that it does this by default, but resorts to alphabetic if this fails?

Comment (bmcutler) [reply site/app/libraries/database/DatabaseQueries.php]: Yes please.   Non-intuitive order better than crash :)

**Comment by bmcutler** [comment]
Please switch the logic for the checks (sorry I told you wrong and it was undocumented)
1 = active course
2 = archived course
(othernumbers) = reserved for other levels of archiving, possibly hidden from instructors, etc.

So make the check 
Your courses, status==1
Your archived courses, status != 1
(and when we expand the usage to other numbers we will revise this page)

**Pull request merged**